### PR TITLE
Electron-401 (fixes crash reporter)

### PR DESCRIPTION
**Pull request by KiranNiranjan**
## Description
Fixes crash dump not being generated on Windows OS [ELECTRON-401](https://perzoinc.atlassian.net/browse/ELECTRON-401)

## Approach
How does this change address the problem?
#### Problem with the code:
 - The crash reporter was initialized before `DOM` ready event
#### Fix:
 - Initializing the crash reporter after `DOM` ready event
 - Initialized crash reporter for `about app window`, `desktop capture window`, `basic auth window` 

## Spectron test results
[Electron-401 — Spectron.pdf](https://github.com/symphonyoss/SymphonyElectron/files/1883622/Electron-401.Spectron.pdf)

## Unit test results
[Electron-401 — Unit Tests.pdf](https://github.com/symphonyoss/SymphonyElectron/files/1883621/Electron-401.Unit.Tests.pdf)

## Open Questions if any and Todos
- [x] Unit-Tests
- [x] Documentation
- [x] Automation-Tests
When solved, check the box and explain the answer.

@VishwasShashidhar , @VikasShashidhar Please review. Thanks 😄

### CODE REVIEW

File: js/crashReporter.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        }
+    }).catch((err) => log.send(
+        logLevels.ERROR,
+        'Unable to initialize crash reporter for main process. Error is ->  ' + err
+    ));
+}
+
+
+/**
+ * Method to initialize crash reporter for render process
```
Comment (VishwasShashidhar) [inline js/crashReporter.js]: should be `renderer` process :)

File: js/crashReporter.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ *
+ * @param browserWindow {Electron.BrowserWindow}
+ * @param extras {Object}
+ */
+function initCrashReporterRenderer(browserWindow, extras) {
+    if (browserWindow && browserWindow.webContents && !browserWindow.isDestroyed()) {
+        getCrashReporterConfig(extras).then((renderCrashReporterData) => {
```
Comment (VishwasShashidhar) [inline js/crashReporter.js]: should be `renderer` :)

**Comment by KiranNiranjan** [comment]
@VishwasShashidhar Added support to generate the crash report. Please Review. Thanks 😄

File: js/crashReporter.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+/**
+ * Method that returns all the required field for crash reporter
+ *
+ * @param extras {object}
+ * @return {Promise<any>}
+ */
+function getCrashReporterConfig(extras) {
+    return new Promise((resolve, reject) => {
+        getMultipleConfigField(['url', 'crashReporter'])
```
Comment (VishwasShashidhar) [inline js/crashReporter.js]: we can probably do this once and store the data in a variable rather than calling the `getMultipleConfigField` every time.

**Comment by VishwasShashidhar** [comment]
lgtm 👍

**Comment by VishwasShashidhar** [comment]
lgtm 👍

**Pull request merged**